### PR TITLE
update previous flaky naive engine test

### LINKED ISSUE #15414: Flaky test_profiler.test_custom_operator_profiling_naive_engine
```
======================================================================

FAIL: test_profiler.test_custom_operator_profiling_naive_engine

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/usr/local/lib/python3.5/dist-packages/nose/case.py", line 198, in runTest

    self.test(*self.arg)

  File "/work/mxnet/tests/python/unittest/test_profiler.py", line 451, in test_custom_operator_profiling_naive_engine

    'test_custom_operator_profiling_multiple_custom_ops_symbolic_naive.json')

  File "/work/mxnet/tests/python/unittest/common.py", line 313, in run_in_spawned_process

    assert p.exitcode == 0, "Non-zero exit code %d from %s()." % (p.exitcode, func.__name__)

AssertionError: Non-zero exit code 255 from test_custom_operator_profiling_multiple_custom_ops_imperative().
```

http://jenkins.mxnet-ci.amazon-ml.com/blue/organizations/jenkins/mxnet-validation%2Funix-cpu/detail/PR-15400/3/pipeline

### PR DESCRIPTION
**Pull request by Zha0q1**
## Description ##
update flaky test reported in https://github.com/apache/incubator-mxnet/issues/15414
fixes #15414
### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by abhinavs95** [comment]
@mxnet-label-bot add [Flaky, test]

File: tests/python/unittest/test_profiler.py [hunk 1]
```diff
@@ -275,6 +275,8 @@ def test_aggregate_duplication():
     file_name = 'test_aggregate_duplication.json'
     enable_profiler(profile_filename = file_name, run=True, continuous_dump=True, \
                     aggregate_stats=True)
+    # clear aggregate stats
+    profiler.dumps(reset = True)
```
Comment (apeforest) [inline tests/python/unittest/test_profiler.py]: nit: remove space around '='. 
FYI: http://google.github.io/styleguide/pyguide.html#36-whitespace

File: tests/python/unittest/test_profiler.py [hunk 2]
```diff
@@ -332,6 +334,8 @@ def create_operator(self, ctx, in_shapes, in_dtypes):
         file_name = 'test_custom_operator_profiling.json'
     enable_profiler(profile_filename = file_name, run=True, continuous_dump=True,\
                     aggregate_stats=True)
+    # clear aggregate stats
+    profiler.dumps(reset = True)
```
Comment (apeforest) [inline tests/python/unittest/test_profiler.py]: nit: remove space around '='. 
FYI: http://google.github.io/styleguide/pyguide.html#36-whitespace

File: tests/python/unittest/test_profiler.py [hunk 3]
```diff
@@ -392,65 +403,44 @@ def infer_shape(self, in_shape):
         def create_operator(self, ctx, shapes, dtypes):
             return MyAdd()
 
-    if file_name is None:
-        file_name = 'test_custom_operator_profiling_multiple_custom_ops_imperative.json'
     enable_profiler(profile_filename = file_name, run=True, continuous_dump=True,\
                     aggregate_stats=True)
+    # clear aggregate stats
+    profiler.dumps(reset = True)
```
Comment (apeforest) [inline tests/python/unittest/test_profiler.py]: nit: remove space around '='. 
FYI: http://google.github.io/styleguide/pyguide.html#36-whitespace

**Comment by Zha0q1** [comment]
@apeforest Can you help me merge? Thanks!

File: tests/python/unittest/test_profiler.py [hunk 4]
```diff
@@ -330,25 +332,34 @@ def create_operator(self, ctx, in_shapes, in_dtypes):
...
-    debug_str = profiler.dumps(format = 'json')
+    debug_str = profiler.dumps(format='json')
     target_dict = json.loads(debug_str)
     assert 'Time' in target_dict and 'Custom Operator' in target_dict['Time'] \
         and 'MySigmoid::pure_python' in target_dict['Time']['Custom Operator'] \
         and '_backward_MySigmoid::pure_python' in target_dict['Time']['Custom Operator'] \
         and 'MySigmoid::_zeros' in target_dict['Time']['Custom Operator']
     profiler.set_state('stop')
 
-def test_custom_operator_profiling_multiple_custom_ops_imperative(seed = None, \
```
Comment (ChaiBapchya) [inline tests/python/unittest/test_profiler.py]: ~~why was function name changed from test_ to check_?~~
Gotcha

**Comment by Zha0q1** [comment]
> LGTM! Add "fixes #issue_number" to description
> So that once its merged, issue gets auto closed.

Added! Thanks!

**Comment by Zha0q1** [comment]
> > LGTM! Add "fixes #issue_number" to description
> > So that once its merged, issue gets auto closed.
> 
> Added! Thanks!

Oh adding that started another round of testing😂

**Pull request merged**